### ISSUE TITLE
Invoice PDF and Shipment PDF does not generate as per storefront  language 

### ISSUE THREAD

**Issue by vajadharmesh** [magento/magento2#19241]
### Preconditions (*)

1.Go to Admin >> SALES >> Invoices >> Click on View >> Download the Invoice PDF

### Steps to reproduce (*)

1.Create multiple stores view from admin, set each store view a different language. each store view display proper language as set.   
2.Now problem is that when you generate invoice PDF its not consider the store view language from where the order was placed.
3.Same problem with Shipment PDF.

### Expected result (*)

1. When you generate invoice PDF its consider the store view language from where the order was placed and translate the content based on that like "Order, Invoice, Shipment" this heading translate based on store view language.

### Actual result (*)

1. Currently it is not prefer the store view language while  translate the PDF's.

**Comment by magento-engcom-team** [comment]
Hi @vajadharmesh. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@vajadharmesh do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by Dharmeshvaja91** [comment]
Hi @magento-engcom-team,  I am working on this issue please  give me $VERSION instance

**Comment by Dharmeshvaja91** [comment]
Hi , @magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @Dharmeshvaja91. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @Dharmeshvaja91, here is your [Magento instance](http://34.228.235.121/i-19241-2-3-develop/).
Admin access: [http://34.228.235.121/i-19241-2-3-develop//admin](http://34.228.235.121/i-19241-2-3-develop//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by Dharmeshvaja91** [comment]
Hi, @magento-engcom-team  please install this  language packs for French and Germany (fr_FR,de_DE) on this instance

**Comment by magento-engcom-team** [comment]
Hi @Dharmeshvaja91. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@Dharmeshvaja91 Thank you for verifying the issue.

Unfortunately, not enough information was provided to created internal ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
@Dharmeshvaja91 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96392` were created

**Comment by sdzhepa** [comment]
re-verified, still can be reproduced on the current `2.3-develop`

**Comment by sdzhepa** [comment]
Hello @vajadharmesh 
fyi: @mautz-et-tong

The internal team is working on this issue right now. In the scope of `MC-35707`.
Fix will be delivered through internal Pull Request

**Comment by LexKoomen** [comment]
This commit is merged into 2.4-develop for this issue: https://github.com/magento/magento2/commit/66d259ed37ac8d7170012e14efb5df12c562cf13

Thought you might wanna know in case you want to create a patch.

**Comment by sdzhepa** [comment]
Closed as fixed and delivered into `2.4-develop`
https://github.com/magento/magento2/issues/19241#issuecomment-678114183
